extensions/json-jackson/src/main/java/io/deephaven/json/jackson/JacksonProvider.java

```diff
@@ -19,7 +33,123 @@
+ * parse a fixed-length JSON array where each element is has the corresponding value type. The values are recursively
+ * resolved and flattened into the {@link ObjectProcessor#outputTypes()}; for example, the earlier example's data could
+ * be re-represented as the JSON array
+ *
+ * <pre>
+ * ["Plymouth", 45.018269, -93.473892]
+ * </pre>
+ * 
+ * and structured as one might expect ({@link TupleValue}({@link StringValue}, {@link DoubleValue},
+ * {@link DoubleValue})), and will produce {@link ObjectProcessor#outputTypes() output types} representing
+ * {@code [String, double, double]}. Even though no field names are present in the JSON value, users may set names for
+ * each element {@link TupleValue#namedValues()} (and will otherwise inherit integer-indexed default names).
+ *
+ * <p>
+ * The {@link TypedObjectValue} represents a union of {@link ObjectValue object values} where the first field is
+ * type-discriminating. For example, the following might be modelled as a type-discriminated object with
+ * type-discriminating field "type", shared "symbol" {@link StringValue}, "quote" object of "bid" {@link DoubleValue}
+ * and an "ask" {@link DoubleValue}, and "trade" object containing a "price" {@link DoubleValue} and a "size"
+ * {@link LongValue}.
+ *
+ * <pre>
+ * {
+ *   "type": "quote",
+ *   "symbol": "BAR",
+ *   "bid": 10.01,
+ *   "ask": 10.05
+ * }
+ * {
+ *   "type": "trade",
+ *   "symbol": "FOO",
+ *   "price": 70.03,
+ *   "size": 42
+ * }
+ * </pre>
+ *
+ * The {@link ObjectProcessor#outputTypes() output types} are first the type-discriminating field, then the shared
+ * fields (if any), followed by the individual {@link ObjectValue object value} fields; with the above example, that
+ * would result in {@link ObjectProcessor#outputTypes() output types}
+ * {@code [String, String, double, double, double long]} and {@link NamedObjectProcessor#names() names}
+ * {@code ["type", "symbol", "quote_bid", "quote_ask", "trade_price", "trade_size"]}.
+ *
+ * <p>
+ * The {@link ObjectKvValue} represents a variable-length object, which expects to parse a JSON object where each
+ * key-value pair has a common {@link ObjectKvValue#value() value type}. The output type will be the output type of the
+ * key and value element types as components types of native arrays ({@link Type#arrayType()}. For example, a JSON
+ * object, whose values are also JSON objects
+ *
+ * <pre>
+ * {
+ *   "Plymouth": {
+ *       "latitude": 45.018269,
+ *       "longitude": -93.473892
+ *   },
+ *   "New York": {
+ *       "latitude": 40.730610,
+ *       "longitude": -73.935242
+ *   }
+ * }
+ * </pre>
+ *
+ * when represented with structuring as one might expect ({@link ObjectKvValue}({@link StringValue},
+ * {@link ObjectValue}({@link DoubleValue}, {@link DoubleValue}))), will produce {@link ObjectProcessor#outputTypes()
+ * output types} representing {@code [String[], double[], double[]]}.
```

Comment:
I don't understand this example, yet.

Reply:
I reworded a small amount. I think it's still a bit hard to understand... that said, I think it is going to be the least used (although, there are some real world cases I've come across making use of it).